scripts/application_to_provider.py

```diff
@@ -255,15 +293,42 @@ def main():
 
         local_storage_dir = secrets.LOCAL_STORAGE if os.path.exists(secrets.LOCAL_STORAGE) else "/tmp"
         local_path = os.path.join(local_storage_dir, sprov_img_uuid)
+
+        def migrate_image_data(img_uuid):
+            # Todo this function is in an awkward place and relies on 'global' state, unsure of best way to refactor
```

Comment:
Yeah this makes `main()` pretty turse -- Let's move this function out and isolate the arguments you require.